overlord/ifacestate/helpers.go

```diff
@@ -377,6 +377,15 @@ func (c *autoConnectChecker) check(plug *interfaces.ConnectedPlug, slot *interfa
 		return false, err
 	}
 
+	var storeAs *asserts.Store
+	if modelAs.Store() != "" {
+		var err error
+		storeAs, err = assertstate.Store(c.st, modelAs.Store())
+		if err != nil && !asserts.IsNotFound(err) {
+			return false, err
+		}
```

Comment:
We are ignoring when err != nil and it is something other than a not found error. Should we handle that here too?

Reply:
I don't understand this comment,  we are ignoring the case when "err is a not found error", not the reverse, because for a bit we don't want to enforce to have store assertions for all stores.

Given that the store assertion can at most allow more this seems ok and allow to land this out of sync with the creation of all store assertion, I expect tough at a later point that we will change snapd to expect the store assertion.

Reply:
Sorry, I said that completely the wrong way. Yes, I was wondering about the case when there is an error and we don't return false, which is, as you correctly pointed out when the "err is a not found error". You answered that question, so marking as resolved.